### ISSUE TITLE
OpenNMT-py Multi Query + Parallel Residual not functioning in CT2

### ISSUE THREAD

**Issue by ArtanisTheOne** [OpenNMT/CTranslate2#1443]
Model trained from scratch in OpenNMT-py converts to CTranslate2 just fine but throws a 'MatMul: k dimension of inputs a and b should match' error on all translation requests. To fix the issue I had to adjust the architecture in OpenNMT-py to turn off multi-query, which made any resulting model work. 
Similarly, turning on parallel_residual in OpenNMT-py throws an error but earlier - in the conversion process. It throws a `KeyError: 'encoder.transformer.0.feed_forward.layer_norm.a_2'` error in the ONMT conversion tool. I assumed parallel_residual was supported because it shows up in the [Decoder class](https://opennmt.net/CTranslate2/python/ctranslate2.specs.TransformerDecoderSpec.html?highlight=parallel%20residual) but please tell me if I'm wrong.

Let me know if more information is required on my part

**Comment by guillaumekln** [comment]
Regarding multi-query, are you using the latest version? The same error was reported and fixed in #1328.

Regarding parallel residual, the parameter is not yet supported in the OpenNMT-py converter. Contributions are welcome to support this parameter.

**Comment by ArtanisTheOne** [comment]
> Regarding multi-query, are you using the latest version? The same error was reported and fixed in #1328.

Yeah I'm on the latest version of both onmt and Ctranslate2. I'm also using `pos_ffn_activation_fn: gelu` like the author of #1328 was. 

> Regarding parallel residual, the parameter is not yet supported in the OpenNMT-py converter

Alright, thank you for the information

**Comment by ArtanisTheOne** [comment]
If it helps I can create a sample model with the problem and send a download link here.

**Comment by guillaumekln** [comment]
Sure.

**Comment by ArtanisTheOne** [comment]
https://drive.google.com/file/d/1vH3GCVdPmDBQMzqIVf5jhVS86BZfbJP5/view?usp=sharing

Here's a zip file containing a converted model with multiquery architecture. It also has the ONMT model in open_nmt_step1.pt if needed. I used some random dummy data, which just had the token "▁d"

**Comment by guillaumekln** [comment]
There is an issue (again) when using multi-query and relative positions. I will fix that.
